### PR TITLE
Add keyboard shortcuts for the prompts on Linux

### LINKED ISSUE #13906: [Linux] `TAB` circulation across different selection buttons (`Cancel/OK/Save`, etc.)
### Check for existing issues

- [X] Completed

### Describe the feature

Currently it does not seem the `TAB` circulation work in linux. Good example is to for example try to `quit` ZED or save current buffer, where pop-up window will show up asking user to `Cancel/Quit` or `Cancel/Don't Save/Save`. Without using mouse/touchpad user currently is unable to select desired selection with `TAB`.

### If applicable, add mockups / screenshots to help present your vision of the feature

Allow `TAB` selection of pop-up window prompts.

### PR DESCRIPTION
**Pull request by hron**
This change adds ability to choose any action from prompts, not just the default one and cancel as Zed has right now. For example, when a user tries to close a file with edits in it the prompt offers "Don't save" option that can be selected only with mouse. Now you can use arrows, tab/shift-tab to pick action and enter/space to confirm it.

Fixes [#13906](https://github.com/zed-industries/zed/issues/13906)

Release Notes:

- Added keyboard navigation in the prompts on Linux ([#13906](https://github.com/zed-industries/zed/issues/13906)).

[Screencast from 2024-07-07 18-02-05.webm](https://github.com/zed-industries/zed/assets/39293/2c7e21b1-cba7-4f5a-92de-6644aefc8a0c)

### CODE REVIEW

File: assets/keymaps/default-macos.json [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
     "bindings": {
       "up": "menu::SelectPrev",
+      "shift-tab": "menu::SelectPrev",
+      "home": "menu::SelectFirst",
       "pageup": "menu::SelectFirst",
       "shift-pageup": "menu::SelectFirst",
       "ctrl-p": "menu::SelectPrev",
       "down": "menu::SelectNext",
+      "tab": "menu::SelectNext",
+      "end": "menu::SelectLast",
```
Comment (mrnugget) [inline assets/keymaps/default-macos.json:13]: Just to be safe: are these idiomatic on macOS?

Comment (hron) [reply assets/keymaps/default-macos.json:13]: no idea. I haven't been using macOS for any significant period of time. I believe it works this way on Windows, so my assumption it should work this way on macOS.

Comment (mikayla-maki) [reply assets/keymaps/default-macos.json:13]: I can confirm that the tab binding is!

**Comment by mrnugget** [comment]
I changed it so that `tab` goes from left to right instead of from right to left, which it did here due to the call to `.rev()` down there. I think left to right for `next` is much more intuitive.

Will merge once CI runs through.

Thank you!

**Pull request merged**